lib/uplinkc/access_scope.go

```diff
@@ -101,6 +104,65 @@ func serialize_scope(scopeRef C.ScopeRef, cerr **C.char) *C.char {
 	return C.CString(serializedScope)
 }
 
+//export restrict_scope
+// restrict_scope restricts a given scope with the provided caveat and encryption restrictions
+func restrict_scope(scopeRef C.ScopeRef, caveat C.Caveat, restrictions **C.EncryptionRestriction, restrictionsLen int, cerr **C.char) C.ScopeRef {
```

Comment:
I don't think we should export a go `int` to a c api. I think `C.size_t` is the appropriate type?

Reply:
Its exported by cgo to header file by default but I agree that `C.size_t` is more appropriate. 

Changed.